src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -550,6 +562,21 @@ result<std::vector<ss::sstring>> service::build_rpk_arguments(
     return rv;
 }
 
+ss::future<> service::cleanup_previous_run() const {
+    if (_rpk_process == nullptr) {
+        co_return;
+    }
+
+    auto& debug_bundle_file = _rpk_process->output_file_path();
+    if (co_await ss::file_exists(debug_bundle_file.native())) {
+        vlog(
+          lg.debug,
+          "Cleaning up previous debug bundle run {}",
+          debug_bundle_file);
+        co_await ss::remove_file(debug_bundle_file.native());
```

Comment:
Probably best if the log is in local filesystem format, but if you prefer your way, either take a copy or hold the process mutex.
```suggestion
    auto debug_bundle_file = _rpk_process->output_file_path().native();
    if (co_await ss::file_exists(debug_bundle_file)) {
        vlog(
          lg.debug,
          "Cleaning up previous debug bundle run {}",
          debug_bundle_file);
        co_await ss::remove_file(debug_bundle_file);
```

Reply:
> hold the process mutex

It's being held by the caller (initiate_rpk_debug_bundle_collection)